src/collection.js

```diff
@@ -652,33 +660,65 @@ export default class Collection {
       syncResultObject.add("errors", data);
     }
 
-    // Handle conflicts, if any
-    await this._handleConflicts(syncResultObject, strategy);
-
     return syncResultObject;
   }
 
-  async importPublications(toApplyLocally) {
-    if (toApplyLocally.length === 0) {
-      return [];
-    }
+  async importPublications(syncResultObject, toApplyLocally, strategy) {
     const toDeleteLocally = toApplyLocally.filter((r) => r.deleted);
     const toUpdateLocally = toApplyLocally.filter((r) => !r.deleted);
 
-    const saved = await this.db.execute((transaction) => {
+    const conflicts = syncResultObject.conflicts;
+
+    const {published, resolved} = await this.db.execute((transaction) => {
       const updated = toUpdateLocally.map((record) => {
         const synced = markSynced(record);
         transaction.update(synced);
-        return {data: synced};
+        return synced;
       });
       const deleted = toDeleteLocally.map((record) => {
         transaction.delete(record.id);
         // Amend result data with the deleted attribute set
-        return {data: {id: record.id, deleted: true}};
+        return {id: record.id, deleted: true};
       });
-      return updated.concat(deleted);
+      const published = updated.concat(deleted);
+      // Handle conflicts, if any
+      const resolved = this._handleConflicts(transaction, conflicts, strategy);
+      return {published, resolved};
+    });
+
+    syncResultObject.add("published", published);
+
+    if (resolved.length > 0) {
+      // Resolved conflicts are merged with previous ones.
+      const resolvedIds = resolved.map((r) => r.id);
+      const mergedResolved = syncResultObject.resolved.filter((r) => resolvedIds.indexOf(r.id) < 0)
+                                                      .concat(resolved);
```

Comment:
Can we do better here? Maybe we should have a `Set` of resolved records, or at least a `Set` of resolved IDs. The indexOf call seems a little ugly to me. Maybe this complexity ought to be in the `SyncResultObject#add` method?